### ISSUE TITLE
Should .parse() default to using process.argv?

### ISSUE THREAD

**Issue by shadowspawn** [tj/commander.js#1103]
I write a lot of test programs where I pass `process.argv` and that is probably the most common calling signature. i.e.

```
program.parse(process.argv);
```

As a convenience, should we default to parsing `process.argv` if parse is called with no parameters?  e.g.

```
program.parse(); // same as calling program.parse(process.argv)
```

The downside is it is a bit magical, but it is up to the user whether they want to take the shortcut. I suggest the examples still pass `process.argv` for clarity.

(I came across this approach in yargs and though it was interesting.)

**Comment by shadowspawn** [comment]
The new behaviour could also include support for the Electron variation in argv (#512). This would  fit into the automatically-do-something-useful intent.

**Comment by shadowspawn** [comment]
Omitting the arguments to `.parse()`and defaulting to `process.argv` is included in the published pre-release of Commander v5.0.0 (#1163)

**Comment by shadowspawn** [comment]
Commander v5.0.0 has been released.

https://github.com/tj/commander.js/releases/tag/v5.0.0
